rust/trezor-thp/src/header.rs

```diff
@@ -0,0 +1,608 @@
+use crate::Role;
+use crate::alternating_bit::SyncBits;
+use crate::control_byte::{self, ControlByte};
+use crate::crc32;
+use crate::error::{self, Error, Result};
+
+use core::marker::PhantomData;
+
+const CHECKSUM_LEN: u16 = crc32::CHECKSUM_LEN as u16;
+const NONCE_LEN: u16 = 8;
+const MAX_PAYLOAD_LEN: u16 = 60000;
+
+const MAX_CHANNEL_ID: u16 = 0xFFEF;
+const BROADCAST_CHANNEL_ID: u16 = 0xFFFF;
+
+/// Represents packet header, i.e. control byte, channel id and possibly payload length.
+/// Please note that `seq_bit` and `ack_bit` which are also part of the header are handled separately.
+#[derive(Clone, Debug, PartialEq)]
+pub enum Header<R: Role> {
+    Continuation {
+        channel_id: u16,
+    },
+    Ack {
+        channel_id: u16,
+        _phantom: PhantomData<R>,
```

Comment:
Just from curiosity - is it a reason for adding it to the `Ack` variant (and not another variant)?

Reply:
Continuation is constructed/patternmatched too often so I moved it to the next one in order not to repeat it as much. Where would you prefer it?